File: app/Models/DataIndikator.php
<?php
namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class DataIndikator extends Model
{
    protected $fillable = [
        'indikator_id',
        'kategori_id',
        'tahun',
        'nilai',
        'satuan',
        'deskripsi'
    ];

    public function kategori()
    {
        return $this->belongsTo(Kategori::class, 'kategori_id', 'id');
    }

    public function indikator()
    {
        return $this->belongsTo(Indikator::class, 'indikator_id', 'id');
    }
}

File: frontend_indikator/lib/models/data_indikator.dart
class DataIndikator {
  final int id;
  final int indikatorId;
  final int kategoriId;
  final int tahun;
 

  DataIndikator({
    required this.id,
    required this.indikatorId,
    required this.kategoriId,
    required this.tahun,
    
    
  });

  factory DataIndikator.fromJson(Map<String, dynamic> json) {
    return DataIndikator(
      id: json['id'],
      indikatorId: json['indikator_id'],
      kategoriId: json['kategori_id'],
      tahun: json['tahun']
      
    );
  }
}
